frontend/src/concepts/pipelines/content/DeletePipelineServerModal.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import DeleteModal from '~/pages/projects/components/DeleteModal';
+import { deletePipelineCR, deleteSecret, getSecret } from '~/api';
+import { getProjectDisplayName } from '~/pages/projects/utils';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+
+type DeletePipelineServerModalProps = {
+  isOpen: boolean;
+  onClose: (deleted: boolean) => void;
+};
+
+const DeletePipelineServerModal: React.FC<DeletePipelineServerModalProps> = ({
+  isOpen,
+  onClose,
+}) => {
+  const [deleting, setDeleting] = React.useState(false);
+  const [error, setError] = React.useState<Error | undefined>();
+  const { project, namespace } = usePipelinesAPI();
+
+  const onBeforeClose = (deleted: boolean) => {
+    onClose(deleted);
+  };
+
+  const deleteName = `${getProjectDisplayName(project)} pipeline server`;
+
+  return (
+    <DeleteModal
+      title="Delete pipeline server?"
+      isOpen={isOpen}
+      onClose={() => onBeforeClose(false)}
+      deleting={deleting}
+      error={error}
+      onDelete={() => {
+        setDeleting(true);
+        Promise.all([
+          new Promise((resolve, reject) => {
+            // if the secret doesn't exist, catch and resolve
+            // else delete the secret, reject if error
+            getSecret(project.metadata.name, 'pipelines-db-password')
+              .then(() =>
+                deleteSecret(project.metadata.name, 'pipelines-db-password')
+                  .then(resolve)
+                  .catch(reject),
+              )
+              .catch(resolve);
+          }),
```

Comment:
@andrewballantyne only thing that is a bit weird imo. For some reason delete secret with dryRun still deleted the secret. so im just getting the secret and only deleting if it exists. Might not be necessary if deleting a non existent secret error is fine

Reply:
Only alternative is instead of `Promise.all` you go with `Promise.allSettled` -- and don't care about failures... 

Any concern about the CR not deleting? Not sure if that can actually ever fail... 🤔

Reply:
i can only image a 404 error for not finding the resource.... which doesnt really matter b/c we just want the secret gone **no matter what**

Reply:
i might switch it just delete secret and catch any failures

Reply:
what do you think about that @andrewballantyne